tests/rust/layered/set_active_layers.rs

```diff
@@ -0,0 +1,147 @@
+    composite::Chorded(composite::Layered(composite::TapHold(
+        layered::ModifierKey::set_active_layers(&[1]),
+    ))),
+    composite::Chorded(composite::Layered(composite::TapHold(
+        layered::ModifierKey::set_active_layers(&[0]),
+    ))),
+    composite::Chorded(composite::LayeredKey::Layered(layered::LayeredKey::new(
+        composite::TapHold(keyboard::Key::new(0x04)),
+        [Some(composite::TapHold(keyboard::Key::new(0x05)))],
+    ))),
+    composite::Chorded(composite::LayeredKey::Layered(layered::LayeredKey::new(
+        composite::TapHold(keyboard::Key::new(0x07)),
+        [Some(composite::TapHold(keyboard::Key::new(0x06)))],
+    ))),
+));
+
+const CONTEXT: Ctx = composite::DEFAULT_CONTEXT;
+
+#[test]
+fn tap_set_active_layers_activates_layers() {
+    // Assemble
+    let mut keymap = Keymap::new(KEYS, CONTEXT);
+    let mut actual_reports = DistinctReports::new();
+
+    // Act
+    // tap the third key, (should be default layer)
+    keymap.handle_input(input::Event::Press { keymap_index: 2 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+
+    keymap.handle_input(input::Event::Release { keymap_index: 2 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+
+    // set layers to [1]
+    keymap.handle_input(input::Event::Press { keymap_index: 0 });
+    keymap.tick();
+    keymap.handle_input(input::Event::Release { keymap_index: 0 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+
+    // tap the two layered keys
+    keymap.handle_input(input::Event::Press { keymap_index: 2 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+
+    keymap.handle_input(input::Event::Release { keymap_index: 2 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+
+    keymap.handle_input(input::Event::Press { keymap_index: 3 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+    keymap.handle_input(input::Event::Release { keymap_index: 3 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+
+    // set layers to [0]
+    keymap.handle_input(input::Event::Press { keymap_index: 1 });
+    keymap.tick();
+    keymap.handle_input(input::Event::Release { keymap_index: 1 });
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+    keymap.tick();
+
+    // tap the second layered keys again
+    keymap.handle_input(input::Event::Press { keymap_index: 3 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
+    keymap.handle_input(input::Event::Release { keymap_index: 3 });
+    keymap.tick();
+    actual_reports.update(keymap.report_output().as_hid_boot_keyboard_report());
```

Comment:
This is unnecessarily difficult to read. -- Better to iterate taps over a slice of indices.